File: String Compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        
        int i = 0;
        int ansIndex = 0;

        while(i < chars.size()) {

            int j = i+1;

            while(j < chars.size() && chars[i] == chars[j]) {
                j++;
            }

            // Yo vector khatm hogya h, ya fer new.different character milgya

            // purana character store kar leta hu
            chars[ansIndex++] = chars[i];

            int count = j-i;

            if(count>1) {
                // adding the count value digit by digit
                string cnt = to_string(count);
                for(char ch: cnt) {
                    chars[ansIndex++] = ch;
                }
            }
            i=j;
        }
        return ansIndex;
    }
};

File: String Compression.java
class Solution {
    public int compress(char[] chars) {
        StringBuffer s = new StringBuffer();
        s.append(chars[0]);
        int count=1;
        for(int i=1;i<chars.length;i++){
            if(chars[i]==chars[i-1]){
                count++;
            }
            else{
                if(count>1){
                    s.append(count);
                    count=1;
                }
                s.append(chars[i]);
            }
        }
        if(count>1){
            s.append(count);
        }
        int n = chars.length;
        String t = s.toString();
        char[] a = t.toCharArray();
        for(int i=0;i<a.length;i++){
            chars[i] = a[i];
        }
        return a.length;
    }
}
